packages/shared/src/components/Feed.spec.tsx

```diff
@@ -1058,3 +1073,126 @@ it('should report irrelevant tags', async () => {
+      .mockImplementation((feat) => {
+        if (feat === feature.userAcquisition) {
+          return mockUserAcquisitionVersion;
+        }
+        return useFeatureOrig(feat);
+      });
+    /* eslint-enable @typescript-eslint/no-var-requires,global-require */
+
+    useRouter.mockImplementation(() => ({
+      route: '/',
+      pathname: '',
+      query: { ua: 'true' },
+      asPath: '',
+      push: jest.fn(),
+      events: {
+        on: jest.fn(),
+        off: jest.fn(),
+      },
+      replace: replaceRouter,
+      beforePopState: jest.fn(() => null),
+      prefetch: jest.fn(() => null),
+    }));
+
+    Object.defineProperty(window, 'location', {
+      value: url,
+      configurable: true,
+    });
+  });
+
+  it('should show the form card when the right url query param is present', async () => {
+    renderComponent([createFeedMock()], defaultUser);
+    await screen.findByTestId('acquisitionFormCard');
+  });
+
+  it('should hide the card once form is submitted', async () => {
+    let mutationCalled = false;
+    renderComponent([createFeedMock()], defaultUser);
+
+    const radio = await screen.findByLabelText('Other');
+    fireEvent.click(radio);
+
+    mockGraphQL({
+      request: {
+        query: USER_ACQUISITION_MUTATION,
+        variables: { acquisitionChannel: 'other' },
+      },
+      result: () => {
+        mutationCalled = true;
+        return { data: { _: null } };
+      },
+    });
+
+    const submit = await screen.findByText('Submit');
+    fireEvent.click(submit);
+
+    await waitFor(() => {
+      expect(mutationCalled).toBeTruthy();
+    });
+
+    await waitFor(() => {
+      expect(replaceRouter).toHaveBeenCalledWith(
+        removeQueryParam(url.toString(), acquisitionKey),
+      );
+    });
+  });
+
+  it('should not show the card if the user dismissed it', async () => {
+    renderComponent([createFeedMock()], defaultUser);
+    const close = await screen.findByLabelText('Close acquisition form');
+    fireEvent.click(close);
+
+    const card = screen.queryByTestId('acquisitionFormCard');
+    expect(card).not.toBeInTheDocument();
+
+    await waitFor(() => {
+      expect(replaceRouter).toHaveBeenCalledWith(
+        removeQueryParam(url.toString(), acquisitionKey),
+      );
+    });
+  });
+
+  it('should not show the card if the user has submitted already', async () => {
+    renderComponent([createFeedMock()], {
+      ...defaultUser,
+      acquisitionChannel: AcquisitionChannel.Blog,
+    });
+
+    const card = screen.queryByTestId('acquisitionFormCard');
+    expect(card).not.toBeInTheDocument();
+  });
+
+  it('should not show the card if the feed is different than My Feed', async () => {
+    renderComponent(undefined, undefined, SharedFeedPage.Popular);
+
+    const card = screen.queryByTestId('acquisitionFormCard');
+    expect(card).not.toBeInTheDocument();
+  });
+
+  it('should not show the card if the userAcquisition feature is in Control version', async () => {
```

Comment:
We usually don't add tests for experiments, but nice to see these.

Reply:
We decided to add the experiment gate only after these tests were already written, so I decided to modify them instead of removing.